components/pages/submission-system/program-submitted-data/index.tsx

```diff
@@ -169,9 +168,46 @@ export default function ProgramSubmittedData({ donorId = '' }: { donorId: string
       },
     });
 
+  const menuItems = clinicalEntityFields.map((entity) => (
+    <VerticalTabs.Item
+      key={entity}
+      active={selectedClinicalEntityTab === aliasedEntityNames[entity]}
+      onClick={(e) => setSelectedClinicalEntityTab(aliasedEntityNames[entity])}
+      disabled={
+        !clinicalData.clinicalEntities.some((e) => e.entityName === aliasedEntityNames[entity])
+      }
+    >
+      {clinicalEntityDisplayNames[entity]}
+      {hasClinicalErrors(clinicalData, entity) && (
+        <VerticalTabs.Tag variant="ERROR">!</VerticalTabs.Tag>
+      )}
+    </VerticalTabs.Item>
+  ));
+
+  const useDefaultQuery =
+    !isFilterUsed &&
+    !currentDonor.length &&
+    (donorPrefixSearch || (searchDonorIds.length === 0 && searchSubmitterIds.length === 0)) &&
+    completionState === 'all';
+
   const noSearchData = searchResultData === null || searchResultData === undefined;
-  const searchResults = noSearchData ? emptySearchResponse : searchResultData;
-  const noData = sideMenuData.clinicalEntities.length === 0 || noSearchData;
+  const noData = clinicalData.clinicalEntities.length === 0 && noSearchData && !currentDonor.length;
+
+  const downloadDonorIds = currentDonor.length
+    ? currentDonor
+    : searchResultData?.clinicalSearchResults?.searchResults.length
+    ? searchResultData.clinicalSearchResults.searchResults.map((result) => result.donorId)
+    : [];
+
+  const downloadSubmitterDonorIds =
+    searchResultData?.clinicalSearchResults?.searchResults
+      .map(({ submitterDonorId }) => submitterDonorId)
+      .filter(Boolean) || [];
```

Comment:
```suggestion
  const downloadSubmitterDonorIds =
    (searchResultData?.clinicalSearchResults?.searchResults || [])
      .map(({ submitterDonorId }) => submitterDonorId)
      .filter(Boolean);
```

you can't iterate on `undefined`

Reply:
Sounds good